ansible/playbooks/roles/kubernetes_master/tasks/main.yml

```diff
@@ -90,8 +88,6 @@
   when: specification.advanced.certificates.renew | bool
   vars:
     valid_days: "{{ specification.advanced.certificates.expiration_days }}"
-    services_to_restart:
```

Comment:
That was done to restart control plane components after certificates generation, so there should be an alternative.

Reply:
Will check that, thanks

Reply:
No need to restart anything more:
```yaml
    certificates:
      expiration_days: 1300
      renew: true
```

```sh
# kubeadm certs check-expiration
CERTIFICATE                EXPIRES                  RESIDUAL TIME   CERTIFICATE AUTHORITY   EXTERNALLY MANAGED
admin.conf                 Aug 31, 2025 10:44 UTC   3y                                      no      
apiserver                  Aug 31, 2025 10:44 UTC   3y              ca                      no      
apiserver-etcd-client      Aug 31, 2025 10:44 UTC   3y              etcd-ca                 no      
apiserver-kubelet-client   Aug 31, 2025 10:44 UTC   3y              ca                      no      
controller-manager.conf    Aug 31, 2025 10:44 UTC   3y                                      no      
etcd-healthcheck-client    Aug 31, 2025 10:44 UTC   3y              etcd-ca                 no      
etcd-peer                  Aug 31, 2025 10:44 UTC   3y              etcd-ca                 no      
etcd-server                Aug 31, 2025 10:44 UTC   3y              etcd-ca                 no      
front-proxy-client         Aug 31, 2025 10:44 UTC   3y              front-proxy-ca          no      
scheduler.conf             Aug 31, 2025 10:45 UTC   3y                                      no      

CERTIFICATE AUTHORITY   EXPIRES                  RESIDUAL TIME   EXTERNALLY MANAGED
ca                      Feb 06, 2032 09:33 UTC   9y              no      
etcd-ca                 Feb 06, 2032 09:33 UTC   9y              no      
front-proxy-ca          Feb 06, 2032 09:33 UTC   9y              no
```

```sh
# kubectl --kubeconfig /etc/kubernetes/admin.conf get pods -A
NAMESPACE              NAME                                                                READY   STATUS    RESTARTS   AGE
kube-system            coredns-6966df8778-l9vh7                                            1/1     Running   0          76m
kube-system            coredns-6966df8778-zchkc                                            1/1     Running   0          76m
kube-system            etcd-rafzei-offlinerhel-kubernetes-master-vm-0                      1/1     Running   0          78m
kube-system            kube-apiserver-rafzei-offlinerhel-kubernetes-master-vm-0            1/1     Running   0          78m
kube-system            kube-controller-manager-rafzei-offlinerhel-kubernetes-master-vm-0   1/1     Running   0          78m
kube-system            kube-flannel-ds-amd64-5xqg6                                         1/1     Running   0          70m
kube-system            kube-flannel-ds-amd64-6ftmw                                         1/1     Running   0          76m
kube-system            kube-proxy-2r6jf                                                    1/1     Running   0          70m
kube-system            kube-proxy-bchml                                                    1/1     Running   0          78m
kube-system            kube-scheduler-rafzei-offlinerhel-kubernetes-master-vm-0            1/1     Running   0          78m
kubernetes-dashboard   dashboard-metrics-scraper-7758c4b7bc-hhgr9                          1/1     Running   0          76m
kubernetes-dashboard   kubernetes-dashboard-56dd58fcb-fqjmd                                1/1     Running   0          76m
```
```sh
# kubectl --kubeconfig /etc/kubernetes/admin.conf get nodes -o wide
NAME                                        STATUS   ROLES                  AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE                                      KERNEL-VERSION           CONTAINER-RUNTIME
rafzei-offlinerhel-kubernetes-master-vm-0   Ready    control-plane,master   78m   v1.22.4   10.1.1.4      <none>        Red Hat Enterprise Linux Server 7.9 (Maipo)   3.10.0-1160.el7.x86_64   containerd://1.4.12
rafzei-offlinerhel-kubernetes-node-vm-0     Ready    <none>                 70m   v1.22.4   10.1.1.5      <none>        Red Hat Enterprise Linux Server 7.9 (Maipo)   3.10.0-1160.el7.x86_64   containerd://1.4.12
```

Reply:
But the docs and cli output stand that the components should be restarted:
```sh
Done renewing certificates. You must restart the kube-apiserver, kube-controller-manager, kube-scheduler and etcd, so that they can use the new certificates.
```
so I will add the containerd restart as well